src/Uno.UI/UI/Xaml/Controls/Grid/Grid.Layout.cs

```diff
@@ -494,7 +494,10 @@ private void LayoutChildren(Span<DoubleRange> calculatedPixelColumns, Span<Doubl
 						totalStarSizedWidth = initialTotalStarSizedWidth - minWidthColumnsStarWidth;
 					}
 
-					unitStarWidth = totalWidth / totalStarSizedWidth;
+					if (totalStarSizedWidth != 0)
```

Comment:
Doh!

There's also a division when the `unitStarWidth` variable is initialized, maybe a similar check is needed there?

Reply:
Indeed. Will open a new issue for this one.